lld/MachO/InputFiles.cpp

```diff
@@ -158,8 +158,10 @@ static bool checkCompatibility(const InputFile *input) {
     return true;
 
   // Swift LLVM fork downstream change start
+  #if defined(__APPLE__)
   error("This version of lld does not support linking for platform " + getPlatformName(platformInfos.front().target.Platform));
```

Comment:
Can we add a warning to say linking macho on via ld64.lld is unsupported and use at your own risk? (for non-apple)